### PR TITLE
Features/generic instruction task

### PR DESCRIPTION
**Pull request by bkb2135**
Task for generic instructions

### CODE REVIEW

File: prompting/tools/datasets/base.py [hunk 1]
```diff
@@ -74,11 +76,51 @@ def next(
...
+            "tags": values,  # tags
+            "extra": {},
+        }
+
+    def random(self, selector: Selector = None):
+        selected = {k: selector(v) for k, v in self.params.items()}
+        return self.get(selected)
+
+    def search(self, params: dict, selector: Selector = None):
+        selected = {k: params.get(k, selector(v)) for k, v in self.params.items()}
```
Comment (steffencruz) [inline prompting/tools/datasets/base.py:125]: Let's make sure all these methods work - do they have test coverage?

File: tests/fixtures/dataset.py [hunk 2]
```diff
@@ -36,7 +36,7 @@
     "content": str,
     "internal_links": list,
     "external_links": list,
-    "source": str,
+    "source": tuple,
```
Comment (steffencruz) [inline tests/fixtures/dataset.py:39]: Why a tuple

File: prompting/tools/datasets/base.py (hunk 1 above)
Comment (bkb2135) [reply prompting/tools/datasets/base.py:125]: Experimentally I believe they work based on testnet runs, but I will add test coverage.

File: tests/fixtures/dataset.py (hunk 2 above)
Comment (bkb2135) [reply tests/fixtures/dataset.py:39]: That's the output given by `info["source"] = (self.__class__.__name__,)`

Comment (steffencruz) [reply tests/fixtures/dataset.py:39]: why is info['source'] a tuple, and not a string?

Comment (bkb2135) [reply tests/fixtures/dataset.py:39]: This is just the way you had it, I can un-tuple it

**Comment by bkb2135** [comment]
**Results of Eval Pipeline**
image

This was taken from a no_weight_setting run on main. 
When grouped by task, this run had in-distribution status codes and rewards (comparing to the Foundations current production run).

File: tests/fixtures/dataset.py (hunk 2 above)
Comment (steffencruz) [reply tests/fixtures/dataset.py]: bahah i don't remember why i did it anymore

File: prompting/utils/config.py [hunk 3]
```diff
@@ -286,7 +286,7 @@ def add_validator_args(cls, parser):
         type=float,
         nargs="+",
         help="The probability of sampling each task.",
-        default=[.25, .25, .25, 0, .25],
+        default=[.2, .2, .2, 0, .2, .2],
```
Comment (steffencruz) [inline prompting/utils/config.py:289]: This should be based on the task registry.
```suggestion
        from prompting.tasks import TASKS
        default=[1/len(TASKS) for task in TASKS],
```

**Review by steffencruz — CHANGES_REQUESTED** [review]
Let's update the config as per my comments and then we can merge pre-staging into staging and run a tracked experiment overnight.

**Review by steffencruz — APPROVED** [review]

**Review by steffencruz — APPROVED** [review]

**Pull request merged**